internal/cluster/validations/validations.go

```diff
@@ -322,7 +338,6 @@ func ParseRegistry(image string) (string, error) {
 // on a list of used images and an ignore list. The ingore list comes as a string and a separator
 // to make it easier to read from a configuration variable
 func getRegistriesWithAuth(ignoreList string, ignoreSeparator string, images ...string) (*map[string]bool, error) {
```

Comment:
Why does this function have to return a map?
It seems like that's making your job in the caller much more difficult.